### PR TITLE
Added Logically Weighted Regression

### PR DESCRIPTION
**Pull request by infiniteoverflow**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

I have implemented a Logically Weighted Regression Model on a given dataset contained in a .csv file.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2000

### CODE REVIEW

**Comment by infiniteoverflow** [comment]
@shraddhavp i have added the program. Kindly let me know if any changes are required. Thank you

**Comment by HarshCasper** [comment]
Hi @infiniteoverflow 

Thanks for making this PR. Please make these changes: 

- Add a line at the End of the File 
- Remove ` .DS_Store` from the Commits 
- Use dashes or underscores in `Logically Weighted Regression`
- Remove the Dataset and add a Kaggle/G-Drive Link instead

**Comment by infiniteoverflow** [comment]
thank you @HarshCasper  , I have made changes accordingly.

File: Python/README.md [hunk 1]
```diff
@@ -199,6 +199,7 @@
 * [Numeric Outlier Selection](ml/Numerical_outlier_method.ipynb)
 * [Principal Component Analysis](ml/principal_component_analysis.ipynb)
 * [Support Vector Machine](ml/Support_Vector_Machine.ipynb)
+* [Logically Weighted Regression](ml/Logically%20Weighted%20Regression)
```
Comment (HarshCasper) [inline Python/README.md]: The Reference is wrong.

File: Python/ml/Logically_weighted_regression.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+
+# Get the dataset from 'https://www.kaggle.com/aswin1871/tips-csv'
```
Comment (HarshCasper) [inline Python/ml/Logically_weighted_regression.py]: Add a beginner pitch about the Algorithm along with the Dataset

File: Python/ml/Logically_weighted_regression.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+def kernel(point,xmat,k):
```
Comment (HarshCasper) [inline Python/ml/Logically_weighted_regression.py:16]: Add proper Docstrings here along with Comments

File: Python/ml/Logically_weighted_regression.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    m,n = np.shape(xmat)
+    weights = np.mat(np.eye(m))
+    for j in range(m):
+        diff = point-xmat[j]
+        weights[j,j] = np.exp((diff*diff.T)/(-2.0*k**2))
+    return weights
+
+
+def local_weight(point,xmat,k,ymat):
```
Comment (HarshCasper) [inline Python/ml/Logically_weighted_regression.py:34]: Add proper Docstrings here along with Comments

File: Python/ml/Logically_weighted_regression.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    wei = kernel(point,xmat,k)
+    w = (xmat.T*(wei*xmat)).I*(xmat.T*(wei*ymat.T))
+    return w
+
+def local_weight_regression(xmat,ymat,k):
```
Comment (HarshCasper) [inline Python/ml/Logically_weighted_regression.py:46]: Add proper Docstrings here along with Comments

File: Python/ml/Logically_weighted_regression.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+x = np.hstack((one.T,mbill.T))
+ypred = local_weight_regression(x,mtip,10)
+
+sortIndex = x[:,1].argsort(0)
+xsort = x[sortIndex][:,0]
+
+plt.scatter(bill,tip,color='green')
+plt.plot(xsort[:,1],ypred[sortIndex],color='red',linewidth=5)
+plt.show()
```
Comment (HarshCasper) [inline Python/ml/Logically_weighted_regression.py]: Enclose this in a `if __name__ == '__main__'` block

**Review by HarshCasper — CHANGES_REQUESTED** [review]
You have not yet removed `.DS_Store` from the Commits

**Comment by infiniteoverflow** [comment]
@HarshCasper I have updated the files as per the requirements. Kindly have a look.
The .DS_STORE file in my commits is not present in my repo , it is pointing the same file in the parent repo.

**Review by HarshCasper — APPROVED** [review]

**Comment by infiniteoverflow** [comment]
@HarshCasper Can you please add the DWOC and Level label for this issue #2000

**Pull request merged**